src/actions/leaderBoardData.js

```diff
@@ -15,6 +16,15 @@ export const getLeaderboardData = userId => {
   };
 };
 
+export const postLeaderboardData = (userId, trophyFollowedUp, userProfile) => {
+  return async dispatch => {
+    const url = ENDPOINTS.TROPHY_ICON(userId, trophyFollowedUp);
+    const res = await httpService.post(url, userProfile);
+
+    await dispatch(postLeaderboardDataAcrionCreator(userProfile));
+  };
+}
+
 export const getOrgData = () => {
 
   return async dispatch => {
```

Comment:
Error handling to debug 

```suggestion
export const postLeaderboardData = (userId, trophyFollowedUp, userProfile) => async dispatch => {
  try {
    const url = ENDPOINTS.TROPHY_ICON(userId, trophyFollowedUp);
    const response = await httpService.post(url, userProfile);
    dispatch(postLeaderboardDataActionCreator(userProfile));
  } catch (error) {
    console.error('Failed to post leaderboard data:', error);
  }
};
```